File: sportsman/ui-web/src/main/webapp/common/services/usermanagement/security-user-service.js
'use strict';

define(['app'], function (app) {

	app.factory('SecurityUserServiceFactory', ['$http', function($http) {
	
	    
	    var SecurityUserServiceFactory = {};
	
	    
	    SecurityUserServiceFactory.createSecurityUser = function (applicationCreationRequest, baseUrl) {
	    	var url = baseUrl + '/SecurityUserService/CreateSecurityUser';
	        return $http.post(url, applicationCreationRequest);
	    };
	    
	    SecurityUserServiceFactory.updateSecurityUser = function (applicationUpdateRequest, baseUrl) {
	    	var url = baseUrl + '/SecurityUserService/UpdateSecurityUser';
	        return $http.post(url, applicationUpdateRequest);
	    };
	    
	    SecurityUserServiceFactory.deleteSecurityUser = function (objectDeletionRequest, baseUrl) {
	    	var url = baseUrl + '/SecurityUserService/DeleteSecurityUser';
	        return $http.post(url, objectDeletionRequest);
	    };
	    
	    SecurityUserServiceFactory.getActiveSecurityUsers = function (baseUrl) {
	    	var url = baseUrl + '/SecurityUserService/GetActiveSecurityUsers';
	        return $http.get(url);
	    };
	    
	    SecurityUserServiceFactory.getSecurityUserById = function (objectRetrievalRequest, baseUrl) {
	    	var url = baseUrl + '/SecurityUserService/GetSecurityUserById';
	    	return $http.post(url, objectRetrievalRequest);
	    };
	    
	    SecurityUserServiceFactory.getSecurityUsers = function (baseUrl) {
	    	var url = baseUrl + '/SecurityUserService/GetSecurityUsers';
	        return $http.get(url);
	    };
	    
	    SecurityUserServiceFactory.getPrefixes = function (baseUrl) {
	    	var url = baseUrl + '/SecurityUserService/GetPrefixes';
	        return $http.get(url);
	    };
	    
	    SecurityUserServiceFactory.updateUserRolesOfSecurityUser = function (securityUserPermissionUpdateRequest, baseUrl) {
	    	var url = baseUrl + '/SecurityUserService/UpdateUserRolesOfSecurityUser';
	        return $http.post(url, securityUserPermissionUpdateRequest);
	    };
	    
	    SecurityUserServiceFactory.updateGrantedPermissionsOfSecurityUser = function (securityUserPermissionUpdateRequest, baseUrl) {
	    	var url = baseUrl + '/SecurityUserService/UpdateGrantedPermissionsOfSecurityUser';
	        return $http.post(url, securityUserPermissionUpdateRequest);
	    };
	    
	    SecurityUserServiceFactory.updateDeniedPermissionsOfSecurityUser = function (securityUserPermissionUpdateRequest, baseUrl) {
	    	var url = baseUrl + '/SecurityUserService/UpdateDeniedPermissionsOfSecurityUser';
	        return $http.post(url, securityUserPermissionUpdateRequest);
	    };
	    
	    SecurityUserServiceFactory.updateSecurityUserProfile = function (securityUserProfileUpdateRequest, baseUrl) {
	    	var url = baseUrl + '/SecurityUserService/UpdateSecurityUserProfile';
	        return $http.post(url, securityUserProfileUpdateRequest);
	    };
	    
	    SecurityUserServiceFactory.getMaritalStatus = function (baseUrl) {
	    	var url = baseUrl + '/SecurityUserService/GetMaritalStatus';
	    	return $http.get(url);
	    };
	    
	
	    return SecurityUserServiceFactory;
	}]);

});

File: sportsman/services-business/src/main/java/com/janaka/projects/services/usermanagement/SecurityUserService.java
package com.janaka.projects.services.usermanagement;

import com.janaka.projects.common.datamanagement.TabularDataRequestModel;
import com.janaka.projects.common.datamanagement.TabularDataResponseModel;
import com.janaka.projects.common.security.AuditContext;
import com.janaka.projects.common.security.SecurityContext;
import com.janaka.projects.dtos.domain.usermanagement.MaritalStatusDTO;
import com.janaka.projects.dtos.domain.usermanagement.PrefixDTO;
import com.janaka.projects.dtos.domain.usermanagement.SecurityUserDTO;
import com.janaka.projects.dtos.requests.common.ObjectDeletionRequest;
import com.janaka.projects.dtos.requests.common.ObjectRetrievalRequest;
import com.janaka.projects.dtos.requests.usermanagement.SecurityUserCreationRequest;
import com.janaka.projects.dtos.requests.usermanagement.SecurityUserPermissionUpdateRequest;
import com.janaka.projects.dtos.requests.usermanagement.SecurityUserProfileUpdateRequest;
import com.janaka.projects.dtos.requests.usermanagement.SecurityUserUpdateRequest;
import com.janaka.projects.dtos.responses.common.ObjectDeletionResponse;
import com.janaka.projects.dtos.responses.common.ObjectListResponse;
import com.janaka.projects.dtos.responses.common.ObjectRetrievalResponse;
import com.janaka.projects.dtos.responses.usermanagement.SecurityUserCreationResponse;
import com.janaka.projects.dtos.responses.usermanagement.SecurityUserUpdateResponse;

public interface SecurityUserService {

  public SecurityUserCreationResponse createSecurityUser(AuditContext auditContext, SecurityContext securityContext,
      SecurityUserCreationRequest request);

  public SecurityUserUpdateResponse updateSecurityUser(AuditContext auditContext, SecurityContext securityContext,
      SecurityUserUpdateRequest request);

  public ObjectDeletionResponse deleteSecurityUser(AuditContext auditContext, SecurityContext securityContext,
      ObjectDeletionRequest request);

  public ObjectListResponse<SecurityUserDTO> getActiveSecurityUsers(AuditContext auditContext,
      SecurityContext securityContext);

  public ObjectRetrievalResponse<SecurityUserDTO> getSecurityUserById(AuditContext auditContext,
      SecurityContext securityContext, ObjectRetrievalRequest request);

  public TabularDataResponseModel<SecurityUserDTO> getSecurityUsers(AuditContext auditContext,
      SecurityContext securityContext, TabularDataRequestModel request);

  public ObjectListResponse<MaritalStatusDTO> getMaritalStatus(AuditContext auditContext,
      SecurityContext securityContext);

  public ObjectListResponse<PrefixDTO> getPrefixes(AuditContext auditContext, SecurityContext securityContext);

  public SecurityUserUpdateResponse updateUserRolesOfSecurityUser(AuditContext auditContext,
      SecurityContext securityContext, SecurityUserPermissionUpdateRequest request);


  public SecurityUserUpdateResponse updateSecurityUserProfile(AuditContext auditContext,
      SecurityContext securityContext, SecurityUserProfileUpdateRequest request);

  public long getActiveCount();


}
